### PR TITLE
#1059 Filter table column based on condition

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes https://github.com/elyra-ai/canvas/issues/1059
Added this new table in `readonlyTable_paramDef.json` -
Screen Shot 2022-06-09 at 12 22 00 PM

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by nmgokhale** [comment]
@matthoward366 One problem in this PR is -
If `"structuretable": [["BP",[20,"low"]], ["Age",[50,"high"]] ]"`. In this case, `"Custom" `column should be visible because there's one row having value `[20,"low"]`. But then the value` [50,"high"]` is hidden. Ideally values in both rows should visible for this case - 

Screen Shot 2022-05-19 at 3 05 46 PM

When `"structuretable": [["BP",[50,"high"]], ["Age",[50,"high"]] ]"`, entire "Custom" column is hidden which is expected in this case - 
Screen Shot 2022-05-19 at 3 11 18 PM

File: canvas_modules/common-canvas/__tests__/test_resources/paramDefs/custom-ctrl-op_paramDef.json [hunk 1]
```diff
@@ -83,6 +98,25 @@
           "type": "string"
         }
       ]
+    },
+    {
```
Comment (matthoward366) [inline canvas_modules/common-canvas/__tests__/test_resources/paramDefs/custom-ctrl-op_paramDef.json:102]: Is there any way to change the values in this table?  If not, do we need it?

Comment (nmgokhale) [reply canvas_modules/common-canvas/__tests__/test_resources/paramDefs/custom-ctrl-op_paramDef.json:102]: In DataStage, this is a readonlytable. They edit the values in `Case Sensitive` column in a tearsheet. But in this PR, Case sensitive column is editable. 

Code works on cell level though. After updating second row, I expected entire column to be hidden but it hides only 2nd row cell - 
hide-col-cell-level

Comment (matthoward366) [reply canvas_modules/common-canvas/__tests__/test_resources/paramDefs/custom-ctrl-op_paramDef.json:102]: Do we really need this custom control/operator?  Can't this just be done with a checkbox that hides the case sensitive column?

Comment (nmgokhale) [reply canvas_modules/common-canvas/__tests__/test_resources/paramDefs/custom-ctrl-op_paramDef.json]: That worked! Removed custom control and added a checkbox in `readonlyTable_paramDef.json` to hide table column - 
Screen Shot 2022-06-09 at 12 22 00 PM

**Review by matthoward366 — APPROVED** [review]
Looks good.

**Pull request merged**